pocketimdb/users/models.py

```diff
@@ -1,7 +1,38 @@
 from django.db import models
-from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+
+class CustomUserManager(BaseUserManager):
+
+    def create_user(self, email, name, password = None, **extra_fields):
+        if not email:
+            raise ValueError('The email must be set')
+        if not name:
+            raise ValueError('The name must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email, name=name, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, name, password = None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+
+        if extra_fields.get('is_staff') is not True:
+            raise ValueError('Superuser must have is_staff=True.')
+        if extra_fields.get('is_superuser') is not True:
+            raise ValueError('Superuser must have is_superuser=True.')
+        return self.create_user(email, name, password, **extra_fields)
```

Comment:
cemu ovo sluzi ?

Reply:
Neophodno mi je da bih mogao da za username polje da koristim email polje. Default-no kreiranje user-a i superuser-a ocekuje da username bude zadat, a ovakvim override-ovanjem tih metoda, dobijam da se vise ne zahteva to username polje.

Reply:
sad vididm da ne mozemo bez ovoga nikako, okej je onda ovako 👍